cmd/kubelet/app/server.go

```diff
@@ -380,6 +380,11 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 		return err
 	}
 
+	kubeClient, err := getKubeClient(s)
```

Comment:
One way (just a suggestion) could be to add an API in the event dispatcher to get a list of currently registered isolators, and then call that from inside the sync loop.